app/lib/url.rb

```diff
@@ -16,7 +16,7 @@ def self.url(uri = nil)
     base_url = "#{protocol}#{domain}"
     return base_url unless uri
 
-    URI.parse(base_url).merge(uri).to_s
+    URI.parse(base_url).merge(Addressable::URI.parse(uri).to_s).to_s
```

Comment:
Seems I uploaded an incomplete version. What I have in my instance now is:
``` ruby
URI.parse(base_url).merge(Addressable::URI.parse(uri).normalize.to_s).to_s
```
but on second thoughts, the optimal code (unless it breaks something else) would be:
``` ruby
Addressable::URI.parse(base_url).join(uri).normalize.to_s
```

Reply:
If all specs are green I'm ok with your optimal version.

Reply:
Now I converted it to my "optimal version" but it failed on two tests that wanted to match dummy URLs created with `URI::Generic` (https://app.travis-ci.com/github/forem/forem/jobs/557611809).

Since they are only places that use `URI::Generic`, can I just have the test code change, or should keep it compatible with `::URI`?

Reply:
Hi, rubocop errors are fixed. Now what should I do with this?

Reply:
Sorry @yheuhtozr, I missed this. If our defaults changed, then it makes sense to update the tests accordingly.